src/app/pages/quickMaps/pages/locationSelect/locationSelect.component.ts

```diff
@@ -36,58 +50,143 @@ export class LocationSelectComponent implements OnInit, AfterViewInit {
+      if (null != country && null != country.geoFeature) {
+        // this.map.fitBounds(country.geoFeature.geometry);
+        this.hoverHighlightFeature(country.geoFeature);
+      }
+    });
+  }
+
+  public getGeoJsonData(): void {
+    this.dictionaryService.getDictionary(DictionaryType.COUNTRIES).then((dict: Dictionary): void => {
+      const featureCollection: GeoJSON.FeatureCollection = {
+        type: 'FeatureCollection',
+        features: dict
+          .getItems<CountryDictionaryItem>()
+          .map((item) => item.geoFeature)
+          .filter((item) => null != item),
+      };
+      this.geojson = L.geoJSON(featureCollection, {
+        style: () => { // eslint-disable-line no-use-before-define
+          return {
+            fillColor: '#8a66ad',
+            fillOpacity: 0.1,
+            color: '#3a1d54',
+            opacity: 0.8,
+          };
+        }
+      });
+    });
+  }
+
+  public getselectedLayer(countryId: string): CountryDictionaryItem {
+    let country: CountryDictionaryItem;
+    if (this.geojson) {
+      this.geojson.eachLayer((layer: any) => {
+        if ((layer.feature.geometry.properties.countryId === countryId)) {
+          country = layer;
+        }
+      });
+      return country;
+    }
+  }
+
+  private initialiseMap(): void {
     this.map = L.map('map').setView([6.6194073, 20.9367017], 3).setMaxZoom(8).setMinZoom(3);
     L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
       attribution: '© <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors',
     }).addTo(this.map);
 
-    void this.http
-      .get('./assets/geoJSON/africanNations.json')
-      .toPromise()
-      .then((json: any) => {
-        this.geojson = L.geoJSON(json, {
-          style: (feature) => {
-            if (feature.properties.sovereignt === 'Malawi') {
-              return {
-                fillColor: 'green',
-                fillOpacity: 0.5,
-              };
-            }
-          },
-          onEachFeature: (feature, layer: L.Layer) => {
-            layer.on({
-              mouseover: () => {
-                this.highlightFeature(layer);
-              },
-              mouseout: () => {
-                this.resetHighlight(layer);
-              },
-              click: (e) => {
-                this.map.fitBounds(e.target.getBounds());
-                window.alert(`you clicked on ${feature.properties.sovereignt}`);
-              },
-            });
-          },
-        }).addTo(this.map);
-      });
+    this.addCountriesMapLayer();
+
+    if (this.activeCountry) {
+      this.selectFeature(this.activeCountry);
+    }
+  }
+
+  private resetHighlight(layer: L.Layer): void {
+    if (layer !== this.selectedCountry) {
+      this.geojson.resetStyle(layer);
+    }
   }
 
   // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-  public highlightFeature(layer: any): void {
+  private hoverHighlightFeature(layer: any): void {
```

Comment:
layer could do with a type.

Reply:
Just wanted to try something, forgot to change it back